diff --git a/packages/core/installer/hack/gen-versions.sh b/packages/core/installer/hack/gen-versions.sh
index 143356b..bcdef01 100755
--- a/packages/core/installer/hack/gen-versions.sh
+++ b/packages/core/installer/hack/gen-versions.sh
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 FIRMWARES="amd-ucode amdgpu-firmware bnx2-bnx2x i915-ucode intel-ice-firmware intel-ucode qlogic-firmware"
-EXTENSIONS="drbd zfs"
+EXTENSIONS="drbd zfs spin tailscale"
 
 talos_version=${1:-$(skopeo --override-os linux --override-arch amd64 list-tags docker://ghcr.io/siderolabs/imager | jq -r '.Tags[]' | grep '^v[0-9]\+.[0-9]\+.[0-9]\+$' | sort -V | tail -n 1)}