diff --git a/packages/core/installer/Makefile b/packages/core/installer/Makefile
index a51e7b61..09117623 100644
--- a/packages/core/installer/Makefile
+++ b/packages/core/installer/Makefile
@@ -34,12 +34,12 @@ image-cozystack:
 	rm -f images/installer.json
 
 image-talos:
-	test -f ../../../_out/assets/installer-amd64.tar || make talos-installer
-	skopeo copy docker-archive:../../../_out/assets/installer-amd64.tar docker://$(REGISTRY)/talos:$(call settag,$(TALOS_VERSION))
+	test -f ../../../_out/assets/installer-arm64.tar || make talos-installer
+	skopeo copy docker-archive:../../../_out/assets/installer-arm64.tar docker://$(REGISTRY)/talos:$(call settag,$(TALOS_VERSION))
 
 image-matchbox:
-	test -f ../../../_out/assets/kernel-amd64 || make talos-kernel
-	test -f ../../../_out/assets/initramfs-metal-amd64.xz || make talos-initramfs
+	test -f ../../../_out/assets/kernel-arm64 || make talos-kernel
+	test -f ../../../_out/assets/initramfs-metal-arm64.xz || make talos-initramfs
 	docker buildx build -f images/matchbox/Dockerfile ../../.. \
 		--tag $(REGISTRY)/matchbox:$(call settag,$(TAG)) \
 		--tag $(REGISTRY)/matchbox:$(call settag,$(TALOS_VERSION)-$(TAG)) \
diff --git a/packages/core/installer/images/matchbox/Dockerfile b/packages/core/installer/images/matchbox/Dockerfile
index 1083ab26..510928e6 100644
--- a/packages/core/installer/images/matchbox/Dockerfile
+++ b/packages/core/installer/images/matchbox/Dockerfile
@@ -1,6 +1,6 @@
-FROM quay.io/poseidon/matchbox:v0.10.0
+FROM quay.io/poseidon/matchbox:v0.11.0-243-gd9e0327a-arm64
 
-COPY _out/assets/initramfs-metal-amd64.xz /var/lib/matchbox/assets/initramfs.xz
-COPY _out/assets/kernel-amd64 /var/lib/matchbox/assets/vmlinuz
+COPY _out/assets/initramfs-metal-arm64.xz /var/lib/matchbox/assets/initramfs.xz
+COPY _out/assets/kernel-arm64 /var/lib/matchbox/assets/vmlinuz
 COPY packages/core/installer/images/matchbox/groups /var/lib/matchbox/groups
 COPY packages/core/installer/images/matchbox/profiles /var/lib/matchbox/profiles
