diff --git a/Makefile b/Makefile
index 514c3d9..b440867 100644
--- a/Makefile
+++ b/Makefile
@@ -35,7 +33,6 @@ build: build-deps
 	make -C packages/system/objectstorage-controller image
 	make -C packages/system/grafana-operator image
 	make -C packages/core/testing image
-	make -C packages/core/talos image
 	make -C packages/core/platform image
 	make -C packages/core/installer image
 	make manifests
