diff --git a/Makefile b/Makefile
index 5041ac6..842a3e7 100644
--- a/Makefile
+++ b/Makefile
@@ -23,6 +23,7 @@ build: build-deps
 	make -C packages/system/cilium image
 	make -C packages/system/linstor image
 	make -C packages/system/linstor-gui image
+	make -C packages/system/kubeovn image
 	make -C packages/system/kubeovn-webhook image
 	make -C packages/system/kubeovn-plunger image
 	make -C packages/system/dashboard image
