diff --git a/packages/core/talos/Makefile b/packages/core/talos/Makefile
index 034b827..fbfe42f 100644
--- a/packages/core/talos/Makefile
+++ b/packages/core/talos/Makefile
@@ -11,12 +11,14 @@ update:
 image: image-matchbox image-talos
 
 image-talos:
-	test -f ../../../_out/assets/installer-arm64.tar || make talos-installer
+	test -f ../../../_out/assets/installer-arm64.tar || $(MAKE) talos-installer
 	skopeo copy docker-archive:../../../_out/assets/installer-arm64.tar docker://$(REGISTRY)/talos:$(call settag,$(TALOS_VERSION))
+	test -f ../../../_out/assets/installer-rpi5-arm64.tar || $(MAKE) talos-installer-rpi5
+	skopeo copy docker-archive:../../../_out/assets/installer-rpi5-arm64.tar docker://$(REGISTRY)/talos:$(call settag,$(TALOS_VERSION))-rpi5
 
 image-matchbox:
-	test -f ../../../_out/assets/kernel-arm64 || make talos-kernel
-	test -f ../../../_out/assets/initramfs-metal-arm64.xz || make talos-initramfs
+	test -f ../../../_out/assets/kernel-arm64 || $(MAKE) talos-kernel
+	test -f ../../../_out/assets/initramfs-metal-arm64.xz || $(MAKE) talos-initramfs
 	docker buildx build -f images/matchbox/Dockerfile ../../.. \
 		--tag $(REGISTRY)/matchbox:$(call settag,$(TAG)) \
 		--tag $(REGISTRY)/matchbox:$(call settag,$(TALOS_VERSION)-$(TAG)) \
@@ -28,27 +30,39 @@ image-matchbox:
 		> ../../extra/bootbox/images/matchbox.tag
 	rm -f images/matchbox.json
 
-assets: talos-iso talos-nocloud talos-metal talos-kernel talos-initramfs
+assets: talos-iso talos-nocloud talos-metal talos-metal-rpi5 talos-kernel talos-initramfs
 
 talos-initramfs:
-	test -f ../../../_out/assets/initramfs-metal-amd64.xz || $(MAKE) build-talos-initramfs
+	test -f ../../../_out/assets/initramfs-metal-arm64.xz || $(MAKE) build-talos-initramfs
 
 talos-kernel:
-	test -f ../../../_out/assets/kernel-amd64 || $(MAKE) build-talos-kernel
+	test -f ../../../_out/assets/kernel-arm64 || $(MAKE) build-talos-kernel
 
 talos-installer:
-	test -f ../../../_out/assets/installer-amd64.tar || $(MAKE) build-talos-installer
+	test -f ../../../_out/assets/installer-arm64.tar || $(MAKE) build-talos-installer
+
+talos-installer-rpi5:
+	test -f ../../../_out/assets/installer-rpi5-arm64.tar || ( \
+		$(MAKE) build-talos-installer-rpi5 && \
+		mv ../../../_out/assets/installer-arm64.tar ../../../_out/assets/installer-rpi5-arm64.tar \
+	)
 
 talos-iso:
-	test -f ../../../_out/assets/metal-amd64.iso || $(MAKE) build-talos-iso
+	test -f ../../../_out/assets/metal-arm64.iso || $(MAKE) build-talos-iso
 
 talos-nocloud:
-	test -f ../../../_out/assets/nocloud-amd64.raw.xz || $(MAKE) build-talos-nocloud
+	test -f ../../../_out/assets/nocloud-arm64.raw.xz || $(MAKE) build-talos-nocloud
 
 talos-metal:
-	test -f ../../../_out/assets/metal-amd64.raw.xz || $(MAKE) build-talos-metal
+	test -f ../../../_out/assets/metal-arm64.raw.xz || $(MAKE) build-talos-metal
+
+talos-metal-rpi5:
+	test -f ../../../_out/assets/metal-rpi5-arm64.raw.xz || ( \
+		$(MAKE) build-talos-metal-rpi5 && \
+		mv ../../../_out/assets/metal-arm64.raw.xz ../../../_out/assets/metal-rpi5-arm64.raw.xz \
+	)
 
-build-talos-initramfs build-talos-kernel build-talos-installer build-talos-iso build-talos-nocloud build-talos-metal:
+build-talos-initramfs build-talos-kernel build-talos-installer build-talos-installer-rpi5 build-talos-iso build-talos-nocloud build-talos-metal build-talos-metal-rpi5:
 	mkdir -p ../../../_out/assets
 	cat images/talos/profiles/$(subst build-talos-,,$@).yaml | \
 		docker run --rm -i -v /dev:/dev --privileged "ghcr.io/siderolabs/imager:$(TALOS_VERSION)" --tar-to-stdout - | \
diff --git a/packages/core/talos/hack/gen-profiles.sh b/packages/core/talos/hack/gen-profiles.sh
index fcad661..6e495b5 100755
--- a/packages/core/talos/hack/gen-profiles.sh
+++ b/packages/core/talos/hack/gen-profiles.sh
@@ -3,7 +3,7 @@ set -e
 set -u
 
 TMPDIR=$(mktemp -d)
-PROFILES="initramfs kernel iso installer nocloud metal"
+PROFILES="initramfs kernel iso installer installer-rpi5 nocloud metal metal-rpi5"
 FIRMWARES="amd-ucode amdgpu bnx2-bnx2x i915 intel-ice-firmware intel-ucode qlogic-firmware"
 EXTENSIONS="drbd zfs spin hailort vc4"
 
@@ -36,6 +36,7 @@ done
 for profile in $PROFILES; do
   echo "writing profile images/talos/profiles/$profile.yaml"
   board=""
+  overlay_block=""
   case "$profile" in
     initramfs|kernel|iso)
       image_options="{}"
@@ -49,6 +50,19 @@ for profile in $PROFILES; do
       platform="metal"
       kind="installer"
       ;;
+    installer-rpi5)
+      image_options="{}"
+      out_format="raw"
+      platform="metal"
+      kind="installer"
+      board="rpi_generic"
+      overlay_block=$(cat <<EOF
+  overlay:
+    imageRef: ghcr.io/siderolabs/sbc-raspberrypi:v0.1.8
+    name: rpi_5
+EOF
+)
+      ;;
     metal)
       image_options="{ diskSize: 1306525696, diskFormat: raw }"
       out_format=".xz"
@@ -56,6 +70,19 @@ for profile in $PROFILES; do
       kind="image"
       board="rpi_generic"
       ;;
+    metal-rpi5)
+      image_options="{ diskSize: 1306525696, diskFormat: raw }"
+      out_format=".xz"
+      platform="metal"
+      kind="image"
+      board="rpi_generic"
+      overlay_block=$(cat <<EOF
+  overlay:
+    imageRef: ghcr.io/siderolabs/sbc-raspberrypi:v0.1.8
+    name: rpi_5
+EOF
+)
+      ;;
     nocloud)
       image_options="{ diskSize: 1306525696, diskFormat: raw }"
       out_format=".xz"
@@ -83,6 +110,7 @@ input:
     path: /usr/install/arm64/initramfs.xz
   baseInstaller:
     imageRef: "ghcr.io/siderolabs/installer:${TALOS_VERSION}"
+${overlay_block}
   systemExtensions:
     - imageRef: ${AMD_UCODE_IMAGE}
     - imageRef: ${AMDGPU_IMAGE}
