diff --git a/Pkgfile b/Pkgfile
index 7dadad0..0beed25 100644
--- a/Pkgfile
+++ b/Pkgfile
@@ -15,7 +15,7 @@ vars:
   UTIL_LINUX_SHA512: 909bba6c38d43ca4cb9bd974c399b4f0537b54709ac43e8acc823d9d9979fbc7523572469eb452b13c86116675d6d2ebb3764444781204f7e5bd0156252635d7
   ENA_VERSION: 2.17.0 # update this when updating PKGS_VERSION in Makefile
   TENSTORRENT_VERSION: 2.8.0 # update this when updating PKGS_VERSION in Makefile
-  HAILORT_VERSION: 4.23.0 # update this when updating PKGS_VERSION in Makefile
+  HAILORT_VERSION: 5.3.0 # update this when updating PKGS_VERSION in Makefile
   MELLANOX_MSFTFLINT_VERSION: 4.34.1-4 # update this when updating PKGS_VERSION in Makefile
   XDMA_VERSION: a78e645 # update this when updating PKGS_VERSION in Makefile
 
diff --git a/drivers/hailort/manifest.yaml.tmpl b/drivers/hailort/manifest.yaml.tmpl
index db4256e..13e4d81 100644
--- a/drivers/hailort/manifest.yaml.tmpl
+++ b/drivers/hailort/manifest.yaml.tmpl
@@ -4,7 +4,7 @@ metadata:
   version: "{{ .VERSION }}"
   author: hailo.ai team
   description: |
-    [{{ .TIER }}] Driver for HailoRT family of AI hardware (eg. Hailo-8L) and is required for PCIe and M.2 Hailo accelerators.
+    [{{ .TIER }}] Driver for HailoRT family of AI hardware (eg. Hailo-8L, Hailo-10H) and is required for PCIe and M.2 Hailo accelerators.
   compatibility:
     talos:
       version: ">= v1.9.0"
diff --git a/drivers/hailort/pkg.yaml b/drivers/hailort/pkg.yaml
index 884fa3a..5d01c58 100644
--- a/drivers/hailort/pkg.yaml
+++ b/drivers/hailort/pkg.yaml
@@ -11,12 +11,10 @@ steps:
       - |
         mkdir -p /rootfs/usr/lib/modules/
         mkdir -p /rootfs/usr/lib/firmware/hailo/
-        mkdir -p /rootfs/usr/lib/udev/rules.d/
 
       - |
         cp -R /usr/lib/modules/* /rootfs/usr/lib/modules/
         cp -R /usr/lib/firmware/hailo/* /rootfs/usr/lib/firmware/hailo/
-        cp -R /usr/lib/udev/rules.d/* /rootfs/usr/lib/udev/rules.d/
     test:
       - |
         mkdir -p /extensions-validator-rootfs
