diff --git a/Pkgfile b/Pkgfile
index a7c24e1..6949841 100644
--- a/Pkgfile
+++ b/Pkgfile
@@ -1,4 +1,4 @@
-# syntax = ghcr.io/siderolabs/bldr:v0.5.6
+# syntax = ghcr.io/siderolabs/bldr:v0.6.0
 
 format: v1alpha2
 
@@ -15,7 +15,7 @@ vars:
   UTIL_LINUX_SHA512: c21ad77b787ab5892169c80cbec1ba46ed6bba36c1db278f2d1cd8712ae237f5cd25bfd20f2dc638334d1c47c5ff6102703147147d42f71c995bd397e735691a
   ENA_VERSION: 2.16.1 # update this when updating PKGS_VERSION in Makefile
   TENSTORRENT_VERSION: 2.7.0 # update this when updating PKGS_VERSION in Makefile
-  HAILORT_VERSION: 4.23.0 # update this when updating PKGS_VERSION in Makefile
+  HAILORT_VERSION: 5.3.0 # update this when updating PKGS_VERSION in Makefile
   MELLANOX_MSFTFLINT_VERSION: 4.34.1-4 # update this when updating PKGS_VERSION in Makefile
 
   # renovate: datasource=git-tags extractVersion=^libtiprc-(?<version>.*)$ depName=git://git.linux-nfs.org/projects/steved/libtirpc.git
diff --git a/drivers/hailort/manifest.yaml.tmpl b/drivers/hailort/manifest.yaml.tmpl
index db4256e..13e4d81 100644
--- a/drivers/hailort/manifest.yaml.tmpl
+++ b/drivers/hailort/manifest.yaml.tmpl
@@ -4,7 +4,7 @@ metadata:
   version: "{{ .VERSION }}"
   author: hailo.ai team
   description: |
-    [{{ .TIER }}] Driver for HailoRT family of AI hardware (eg. Hailo-8L) and is required for PCIe and M.2 Hailo accelerators.
+    [{{ .TIER }}] Driver for HailoRT family of AI hardware (eg. Hailo-8L, Hailo-10H) and is required for PCIe and M.2 Hailo accelerators.
   compatibility:
     talos:
       version: ">= v1.9.0"
diff --git a/drivers/hailort/pkg.yaml b/drivers/hailort/pkg.yaml
index 884fa3a..5d01c58 100644
--- a/drivers/hailort/pkg.yaml
+++ b/drivers/hailort/pkg.yaml
@@ -11,12 +11,10 @@ steps:
       - |
         mkdir -p /rootfs/usr/lib/modules/
         mkdir -p /rootfs/usr/lib/firmware/hailo/
-        mkdir -p /rootfs/usr/lib/udev/rules.d/
 
       - |
         cp -R /usr/lib/modules/* /rootfs/usr/lib/modules/
         cp -R /usr/lib/firmware/hailo/* /rootfs/usr/lib/firmware/hailo/
-        cp -R /usr/lib/udev/rules.d/* /rootfs/usr/lib/udev/rules.d/
     test:
       - |
         mkdir -p /extensions-validator-rootfs
